src/types/instantsearch.ts

```diff
@@ -91,11 +92,26 @@ export type NumericRefinement = {
 
 export type Refinement = FacetRefinement | NumericRefinement;
 
+export type Router = {
+  onUpdate(callback: (route: Partial<SearchParameters>) => void): void;
+  read(): UiState;
+  write(route: Partial<SearchParameters>): void;
+  createURL(state: UiState): string;
+  dispose(): Widget['dispose'];
+};
+
+export type StateMapping = {
```

Comment:
This should probably be an interface that is implemented by the `simple` mapping.

Reply:
Yes – that's for another PR.